app/models/project_media.rb

```diff
@@ -250,48 +250,51 @@ def replace_by(new_pm)
       raise I18n.t(:replace_blank_media_only)
     else
       # Save the new item
-      analysis = self.analysis
       new_pm.updated_at = Time.now
       new_pm.skip_check_ability = true
       new_pm.channel = { main: CheckChannels::ChannelCodes::FETCH }
       new_pm.save(validate: false) # To skip channel validation
-      new_pm.analysis = { title: analysis['title'], content: analysis['content'] }
-      # Point the claim
+
+      # Point the claim and consequently the fact-check
+      new_pm = ProjectMedia.find(new_pm.id)
       new_pm.claim_description = self.claim_description
+      new_pm.save!
+
+      # All annotations from the old item should point to the new item
+      # Remove any status and report from the new item
+      Annotation.where(annotation_type: ['verification_status', 'report_design'], annotated_type: 'ProjectMedia', annotated_id: new_pm.id).delete_all
+      Annotation.where(annotated_type: 'ProjectMedia', annotated_id: self.id).update_all(annotated_id: new_pm.id)
+
+      # All versions from the old item should point to the new item
+      Version.from_partition(self.team_id).where(associated_type: 'ProjectMedia', associated_id: self.id).update_all(annotated_id: new_pm.id)
```

Comment:
Should update associated_id and for the actions related to ProjectMedia we should also update item_id

Reply:
Thanks Sawy, done.